cms/models.py

```diff
@@ -2635,3 +2638,126 @@ def get_context(self, request, *args, **kwargs):
                 "HUBSPOT_ENTERPRISE_PAGE_FORM_ID"
             ),
         }
+
+
+class CommonComponentIndexPage(CanCreatePageMixin, DisableSitemapURLMixin, Page):
+    """
+    A placeholder class to group CommonChildPages as children.
+    This class logically acts as no more than a "folder" to organize
+    pages and add parent slug segment to the page url.
+    """
+
+    slug = COMMON_COURSEWARE_COMPONENT_INDEX_SLUG
+
+    parent_page_types = ["HomePage"]
+
+    subpage_types = [
+        "ForTeamsCommonPage",
+        "LearningTechniquesCommonPage",
+    ]
+
+    # disable promote panels, no need for slug entry, it will be autogenerated
+    promote_panels = []
+
+    def serve(self, request, *args, **kwargs):  # noqa: ARG002
+        """
+        For index pages we raise a 404 because these pages do not have a template
+        of their own and we do not expect a page to available at their slug.
+        """
+        raise Http404
+
+
+class CommonChildPageMixin(models.Model):
+    """
+    Abstract model for common child pages associated with a platform.
+
+    Attributes:
+        platform (ForeignKey): Optional reference to a platform for the page.
+    """
+
+    platform = models.ForeignKey(
+        Platform, on_delete=models.SET_NULL, null=True, blank=True
+    )
+
+    parent_page_types = [
+        "CommonComponentIndexPage",
+    ]
+
+    # disable promote panels, no need for slug entry, it will be autogenerated
+    promote_panels = []
+
+    class Meta:
+        abstract = True
+
+    def __str__(self):
+        return f"{self.title} - {self.platform}" if self.platform else self.title
+
+    def save(self, clean=True, user=None, log_action=False, **kwargs):  # noqa: FBT002
+        # autogenerate a unique slug so we don't hit a ValidationError
+        if not self.title:
+            self.title = self.__class__._meta.verbose_name.title()  # noqa: SLF001
+        self.slug = slugify(f"{self.get_parent().id}-{self.title}-{self.platform}")
+        super().save(clean=clean, user=user, log_action=log_action, **kwargs)
```

Comment:
This will call the save from `CourseProgramChildPage`, and we will end up with slug `self.slug = slugify(f"{self.get_parent().id}-{self.title}")`. This will prevent us from creating 2 pages with same title.

You can do this by creating 2 same pages for different platforms with same title.